### ISSUE TITLE
No automatic `nearest` interpolation for masks in `RandomElasticTransform`

### ISSUE THREAD

**Issue by JanSellner** [kornia/kornia#2002]
### Describe the bug

Usually, the data type `DataKey.MASK` uses the `nearest` interpolation method to preserve label index values (e.g. when using `RandomAffine`). However, this does not work for the `RandomElasticTransform` augmentation since the corresponding parameter is named `mode` and not `resample`. Hence the default argument to `AugmentationSequential`
```python
extra_args={DataKey.MASK: dict(resample=Resample.NEAREST, align_corners=True)}
```
does not trigger the `nearest` interpolation mode.

### Reproduction steps

```bash
Here is an example reproducing the issue of the current default interpolation for mask input types:

import torch
import kornia.augmentation as K
features = torch.rand(1, 1, 4, 4)
labels = torch.ones(1, 1, 4, 4, dtype=torch.float32) * 10
labels[:, :, :, :2] = 0
labels[:, :, :2, :] = 0
# tensor([[[[ 0.,  0.,  0.,  0.],
#           [ 0.,  0.,  0.,  0.],
#           [ 0.,  0., 10., 10.],
#           [ 0.,  0., 10., 10.]]]])

compose = K.AugmentationSequential(K.RandomElasticTransform(alpha=(10, 10)))
torch.manual_seed(0)
compose(features, labels, data_keys=["input", "mask"])[1]
# tensor([[[[0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00],
#           [0.0000e+00, 0.0000e+00, 4.4417e-03, 4.3297e-03],
#           [0.0000e+00, 0.0000e+00, 9.8832e+00, 1.0000e+01],
#           [0.0000e+00, 0.0000e+00, 9.8833e+00, 1.0000e+01]]]])
```

### Expected behavior

The default behavior for the elastic transformation should be similar to the other transformations, i.e. `nearest` interpolation for mask types.

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
- PyTorch Version (e.g., 1.0): 1.13
- OS (e.g., Linux): Linux
- How you installed PyTorch (`conda`, `pip`, source): pip
- Build command you used (if compiling from source):
- Python version: 3.10.4
- CUDA/cuDNN version: 11.7
- GPU models and configuration: 3090
- Any other relevant information:
```


### Additional context

Current workaround for this issue is to manually set the interpolation mode:
```python
compose = K.AugmentationSequential(*transforms, extra_args={DataKey.MASK: dict(resample=Resample.NEAREST, align_corners=True, mode="nearest")})
```

One option would be to rename the `mode` parameter to `resample`, than the default mask handling triggers as well. This would be a breaking change but I see that elastic transformation might get some exciting new functionality in the future anyway (issue #994) so this could be a combined change.

**Comment by edgarriba** [comment]
@shijianjian

**Comment by shijianjian** [comment]
@JanSellner I believe this is a mistake for `RandomElasticTransform`. Can you send a PR to deprecate the mode argument and to use resample as the default?
